test/extended/util/framework.go

```diff
@@ -390,13 +390,19 @@ func (t *BuildResult) dumpRegistryLogs() {
 	}
 
 	since := time.Now().Sub(*buildStarted)
+
+	// Changing the namespace on the derived client still changes it on the original client
+	// because the kubeFramework field is only copied by reference. Saving the original namespace
+	// here so we can restore it when done with registry logs
+	savedNamespace := t.oc.Namespace()
 	oadm := t.oc.AsAdmin().SetNamespace("default")
 	out, err := oadm.Run("logs").Args("dc/docker-registry", "--since="+since.String()).Output()
 	if err != nil {
 		fmt.Fprintf(g.GinkgoWriter, "Error during log retrieval: %+v\n", err)
 	} else {
 		fmt.Fprintf(g.GinkgoWriter, "%s\n", out)
 	}
+	t.oc.SetNamespace(savedNamespace)
```

Comment:
not AsAdmin?  couldn't there be cases where the namespace is only accessible as admin?